src/orion/serving/storage_resource.py

```diff
@@ -0,0 +1,232 @@
+
+    def set_progress(self, message: str, progress: float):
+        self.progress_message.value = message.encode()
+        self.progress_value.value = progress
+
+    def is_completed(self):
+        """Return True if task is completed"""
+        return self.completed.value
+
+    def set_completed(self, success=True):
+        """Set task terminated status"""
+        self.completed.value = 1 if success else -1
+
+
+def _import_data(task: ImportTask, storage, load_host, resolve, name, version):
+    """Function to run import task.
+
+    Set stream handler, launch load_database and set task status.
+    """
+    try:
+        print("Import starting.", task.task_id)
+        logging.basicConfig(stream=task.notifications, force=True, level=logging.INFO)
+        load_database(
+            storage,
+            load_host,
+            resolve,
+            name,
+            version,
+            progress_callback=task.set_progress,
+        )
+        task.set_completed(success=True)
+    except Exception as exc:
+        traceback.print_tb(exc.__traceback__)
+        print("Import error.", exc)
+        # Add error message to shared messages
+        task.notifications.write(f"Error: {exc}")
+        task.set_completed(success=False)
+    finally:
+        # Remove imported files
+        os.unlink(load_host)
+        lock_file = f"{load_host}.lock"
+        if os.path.exists(lock_file):
+            os.unlink(lock_file)
+        print("Import terminated.")
+
+
+class StorageResource:
+    """Handle requests for the dump/ REST endpoint"""
+
+    def __init__(self, storage):
+        self.storage = storage
+        self.current_task: ImportTask = None
+
+    def on_get_dump(self, req: Request, resp: Response):
+        """Handle the GET requests for dump/"""
+        name = req.get_param("name")
+        version = req.get_param_as_int("version")
+        dump_host = _gen_host_file(basename="dump")
+        download_suffix = "" if name is None else f" {name}"
+        if download_suffix and version is not None:
+            download_suffix = f"{download_suffix}.{version}"
+        try:
+            dump_database(self.storage, dump_host, name=name, version=version)
+            resp.downloadable_as = f"dump{download_suffix if download_suffix else ''} ({datetime.now()}).pkl"
+            resp.content_type = "application/octet-stream"
+            with open(dump_host, "rb") as file:
+                resp.data = file.read()
+        except DatabaseError as exc:
+            raise falcon.HTTPNotFound(title=type(exc).__name__, description=str(exc))
+        finally:
+            os.unlink(dump_host)
+            os.unlink(f"{dump_host}.lock")
+
+    def on_post_load(self, req: Request, resp: Response):
+        """Handle the POST requests for load/"""
+        if self.current_task is not None and not self.current_task.is_completed():
+            raise falcon.HTTPForbidden(description="An import is already running")
```

Comment:
We would perhaps add an option to allow cancelling an import. If it's too much work you could create an issue instead and delay this work.

Reply:
Ok ! I opened an issue: https://github.com/Epistimio/orion/issues/1085